### PR TITLE
Research costs based on multiple items/tags [1.20]

### PR DESCRIPTION
**Pull request by Thodor12**
See #9490

### CODE REVIEW

File: src/api/java/com/minecolonies/api/research/AbstractResearchProvider.java [hunk 1]
```diff
@@ -532,42 +537,83 @@ public Research addResearchRequirement(final ResourceLocation researchReq, final
...
+
+        /**
+         * Adds an item cost to the research. This will be consumed when beginning the research, and will not be refunded.
+         * Multiple ItemCosts are supported, but for UI reasons it's encouraged to keep to 5 or less.
+         *
+         * @param items The item to require.
+         * @param count The number of the item to require.
+         * @return this.
+         */
+        public Research addItemCost(final List<Item> items, final int count)
```
Comment (someaddons) [inline src/api/java/com/minecolonies/api/research/AbstractResearchProvider.java:560]: is there a reason we're using Items instead of ItemStacks?

Comment (Thodor12) [reply src/api/java/com/minecolonies/api/research/AbstractResearchProvider.java:560]: This previous ItemStack one was not being used. So I did not bother continuing it either, I only kept the item/count one and added the tag/list ones.

File: src/main/java/com/minecolonies/coremod/client/gui/WindowResearchTree.java [hunk 2]
```diff
@@ -961,27 +962,16 @@ else if (requirement instanceof BuildingResearchRequirement buildingRequirement)
...
-            icon.setItem(is);
+            final RotatingItemIcon icon = new RotatingItemIcon();
             icon.setPosition(offsetX + RESEARCH_WIDTH - storageXOffset - INITIAL_X_OFFSET, offsetY + NAME_LABEL_HEIGHT + TEXT_Y_OFFSET);
             icon.setSize(DEFAULT_COST_SIZE, DEFAULT_COST_SIZE);
+            icon.setItems(cost.getItems().stream().map(ItemStack::new).map(stack -> {
+                stack.setCount(cost.getCount());
+                return stack;
+            }).toList());
             view.addChild(icon);
-            if((InventoryUtils.getItemCountInItemHandler(new InvWrapper(Minecraft.getInstance().player.getInventory()),
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowResearchTree.java:974]: No more highlighting?

File: src/main/java/com/minecolonies/coremod/research/costs/CostUtils.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.minecolonies.coremod.research.costs;
+
+import com.google.gson.JsonObject;
+
+import static com.minecolonies.coremod.research.GlobalResearch.RESEARCH_QUANTITY_PROP;
+
+/**
+ * Util methods for cost entries.
+ */
+public class CostUtils
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/research/costs/CostUtils.java]: Could add this to the interface, maybe even as a default method.

**Review by someaddons — APPROVED** [review]

**Pull request merged**